Update README.md, add architecture.md  and a new diagram (#110)
kyma-project/runtime-watcher: docs/05-10-api.md

# Watcher CR

> The [Watcher CR](https://github.com/kyma-project/lifecycle-manager/blob/main/api/v1beta2/watcher_types.go#L121) configures Kyma Control Plane (KCP) setup and Runtime Watcher on Kyma runtimes. Read the document to see the the parameters the CR consists of.

Reviewer: ```suggestion
The [Watcher CR](https://github.com/kyma-project/lifecycle-manager/blob/main/api/v1beta2/watcher_types.go#L121) configures the Kyma Control Plane (KCP) setup and Runtime Watcher on Kyma runtimes. Read the following document to see the the parameters the CR consists of.
```

## Runtime-Watcher configuration

### **spec.resourceToWatch**

> Defines the resources to watch by Runtime Watcher. By default, the **version** value is a wildcard (`*`) so an update is not required after the API version of a kind changes.

Reviewer: ```suggestion
Defines which resources the Runtime Watcher watches. By default, the **version** value is a wildcard (`*`), so an update is not required after the API version changes.
```

If we can drop the "of a kind" without losing meaning, it would be much easier to read.

```yaml
spec:
  resourceToWatch:
    group: operator.kyma-project.io
    version: "*"
    resource: kymas
```

### **spec.labelsToWatch**

> > **NOTE:** The resources you're watching need to have the `operator.kyma-project.io/watched-by` label. The label also informs where to route requests to.

Reviewer: ```suggestion
> **NOTE:** The resources you're watching must have the `operator.kyma-project.io/watched-by` label. The label also informs where to route requests to.
```

> The **spec.labelsToWatch** attribute is used to be able to filter the specified Group/Version/Kind (GVK) in **spec.resourceToWatch**. For example, if the specified GVK is `secrets`, then it is useful to filter them by a specific label, otherwise, the Runtime Watcher would send an event to the KCP for every Create/Read/Update/Delete (CRUD) event of any Secret in the Kyma cluster.

Reviewer: Can we simplify this verb phrase?

```suggestion
With the **spec.labelsToWatch** attribute, you can filter the specified Group/Version/Kind (GVK) in **spec.resourceToWatch**. For example, if the specified GVK is `secrets`, then it is useful to filter them by a specific label, otherwise, the Runtime Watcher would send an event to the KCP for every Create/Read/Update/Delete (CRUD) event of any Secret in the Kyma cluster.
```

```yaml
spec:
  labelsToWatch:
    "operator.kyma-project.io/watched-by": "lifecycle-manager"
    "example.label.to.watch": "true"
  resourceToWatch:
    group: ""
    version: "v1"
    resource: secrets
```

### **spec.field**

> It uses either the `spec` or `status` value to define if Runtime Watcher should send an event to KCP if the spec of the specified GVK or the status changes.

Reviewer: ```suggestion
It uses either the `spec` or `status` value to define if Runtime Watcher sends an event to KCP if the spec of the specified GVK or the status changes.
```
[Trying to avoid "should"](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#terminology).

## KCP configuration

### **spec.serviceInfo**

> Specifies the name and Namespace, as well as the port, the incoming events should be routed to.

Reviewer: ```suggestion
Specifies the name and Namespace, as well as the port, the incoming events are routed to.
```
[Trying to avoid "should"](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#terminology).

```yaml
spec:
    name: klm-event-service
    namespace: kcp-system
    port: 8082
```

### **spec.gateway**

> Defines the label selector of a Gateway to which the VirtualService should bind to.

Reviewer: ```suggestion
Defines to which label selector of a Gateway the VirtualService binds.
```
[Trying to avoid "should"](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#terminology).
